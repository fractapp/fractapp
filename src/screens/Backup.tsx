import React, {useContext, useEffect} from 'react';
import {
  StyleSheet,
  View,
  ScrollView,
  Text,
  FlatList,
  Image,
  TouchableOpacity,
} from 'react-native';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import Dialog from 'storage/Dialog';
import BackupUtils from 'utils/backup';
import db from 'utils/db';
import backupUtil from 'utils/backup';

export const Backup = ({navigation, route}: {navigation: any; route: any}) => {
  const dialogContext = useContext(Dialog.Context);

  const onSuccess = () => {
    navigation.reset({
      index: 0,
      routes: [{name: 'Home'}],
    });
    dialogContext.dispatch(Dialog.close());
  };

  useEffect(() => {
    if (route.params?.isSuccess == null) {
      return;
    }

    const dir =
      route.params.type == BackupUtils.BackupType.File
        ? 'Downloads'
        : backupUtil.GoogleDriveFolder;

    dialogContext.dispatch(
      Dialog.open(
        'Success save wallet',
        `If you lose access to file then you will not be able to restore access to the wallet. File "${route.params?.fileName}.json" saved in "${dir}" directory`,
        onSuccess,
      ),
    );
  }, [route.params?.isSuccess]);

  const backupFile = async () => {
    const seed = await db.getSeed();
    await BackupUtils.backupFile(
      () =>
        navigation.navigate('WalletFileBackup', {
          seed: seed?.split(' '),
          type: BackupUtils.BackupType.File,
          callerScreen: 'Backup',
        }),
      () =>
        dialogContext.dispatch(
          Dialog.open(
            'Open settings',
            'If you want to save a file then open the application settings and give it access to the storage.',
            () => dialogContext.dispatch(Dialog.close()),
          ),
        ),
    );
  };
  const backupGoogleDrive = async () => {
    const seed = await db.getSeed();
    await BackupUtils.backupGoogleDrive(() =>
      navigation.navigate('WalletFileBackup', {
        seed: seed?.split(' '),
        type: BackupUtils.BackupType.GoogleDrive,
        callerScreen: 'Backup',
      }),
    );
  };
  const backupSeed = async () => {
    const seed = await db.getSeed();
    navigation.navigate('SaveSeed', {
      seed: seed?.split(' '),
      onSuccess: () =>
        navigation.reset({
          index: 0,
          routes: [{name: 'Home'}],
        }),
    });
  };

  const menuItems = [
    {
      img: (
        <Image
          source={require('assets/img/google-drive.png')}
          style={{width: 32, height: 32}}
        />
      ),
      title: 'Google disk',
      onClick: () => backupGoogleDrive(),
    },
    {
      img: <MaterialIcons name="folder-open" size={32} color="#888888" />,
      title: 'File to phone',
      onClick: () => backupFile(),
    },
    {
      img: (
        <MaterialCommunityIcons name="content-copy" size={32} color="#888888" />
      ),
      title: 'Backup seed',
      onClick: () => backupSeed(),
    },
  ];
  const renderItem = ({item}: {item: any}) => (
    <TouchableOpacity style={styles.menuItem} onPress={item.onClick}>
      {item.img}
      <Text style={styles.menuTitle}>{item.title}</Text>
    </TouchableOpacity>
  );

  return (
    <View style={styles.profile}>
      <ScrollView showsVerticalScrollIndicator={false} style={{width: '100%'}}>
        <View style={{alignItems: 'center', justifyContent: 'center'}}>
          <Text style={styles.description}>
            Save your wallet in a safe place. If you lose your wallet, you
            cannot restore access to it.
          </Text>
          <FlatList
            style={styles.menu}
            ItemSeparatorComponent={() => <View style={styles.dividingLine} />}
            data={menuItems}
            renderItem={renderItem}
            keyExtractor={(item) => item.title}
          />
        </View>
      </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  profile: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  dividingLine: {
    alignSelf: 'center',
    width: '100%',
    borderBottomWidth: 1,
    borderBottomColor: '#DADADA',
  },
  description: {
    width: '90%',
    marginTop: 10,
    fontSize: 15,
    fontFamily: 'Roboto-Regular',
    color: '#888888',
  },
  menu: {
    flex: 1,
    width: '88%',
    marginTop: 20,
  },
  menuItem: {
    flexDirection: 'row',
    paddingTop: 10,
    paddingBottom: 10,
  },
  menuTitle: {
    marginLeft: 10,
    fontSize: 17,
    fontFamily: 'Roboto-Regular',
    fontStyle: 'normal',
    color: 'black',
    alignSelf: 'center',
  },
});
