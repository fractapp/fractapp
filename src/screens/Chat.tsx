import React, { useEffect, useState } from 'react';
import { FlatList, Image, Linking, StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import { WalletLogo } from 'components/WalletLogo';
import ChatsStore from 'storage/Chats';
import GlobalStore from 'storage/Global';
import stringUtils from 'utils/string';
import StringUtils from 'utils/string';
import backend from 'utils/api';
import { Button, Message } from 'types/message';
import { MessageView } from 'components/MessageView';
import { AddressOnly, Profile, User } from 'types/profile';
import { PaymentMsg } from 'components/PaymentMsg';
import { Currency, getNetwork, toCurrency } from 'types/wallet';
import { randomAsHex } from '@polkadot/util-crypto';
import { useDispatch, useSelector } from 'react-redux';
import UsersStore from 'storage/Users';
import DialogStore from 'storage/Dialog';
import AccountsStore from 'storage/Accounts';
import { Adaptors } from 'adaptors/adaptor';
import { AccountType, Network } from 'types/account';
import ServerInfoStore from 'storage/ServerInfo';

/**
 * Chat screen
 * @category Screens
 */
export const Chat = ({navigation, route}: {navigation: any; route: any}) => {
  const globalState: GlobalStore.State = useSelector((state: any) => state.global);
  const usersState: UsersStore.State = useSelector((state: any) => state.users);
  const accountsState: AccountsStore.State = useSelector((state: any) => state.accounts);
  const chatsState: ChatsStore.State = useSelector((state: any) => state.chats);
  const serverInfoState: ServerInfoStore.State = useSelector((state: any) => state.serverInfo);

  const dispatch = useDispatch();

  const chatId: string = route.params.chatId;

  const chatInfo = chatsState.chatsInfo[chatId]!;
  const user: User = usersState.users[chatId]!;

  const [notificationCount] = useState(chatInfo.notificationCount);
  const [messages, setMessages] = useState<Array<Message>>([]);
  const [lengthOffset, setLengthOffset] = useState<number>(0);

  const onPressChatBtn = async (msgId: string, btn: Button) => {
    switch (btn.action) {
      case '/enterAmount':
        if (btn.arguments.length < 3) {
          return;
        }

        const currencyEnterAmount: Currency = toCurrency(btn.arguments[0]);

        navigation.navigate('EnterAmount', {
          chatId: chatId,
          msgId: msgId,
          isChatBotRequest: true,
          isUSDMode: true,
          value: '',
          currency: currencyEnterAmount,
          args: btn.arguments.slice(1, btn.arguments.length),
        });
        break;
      case '/openUrl':
        if (btn.arguments.length < 1) {
          return;
        }
        await Linking.openURL(btn.arguments[0]); //TODO: next release alert (any site)
        break;
      case '/broadcast':
        try {
          if (btn.arguments.length < 2) {
            return;
          }

          const tx = JSON.parse(btn.arguments[0]);
          const currency: Currency = toCurrency(btn.arguments[1]);

          const network: Network = getNetwork(currency);
          const api = Adaptors.get(network)!;

          dispatch(DialogStore.actions.showConfirmTxInfo(
            await api.parseTx((user.value as Profile),
              accountsState.accounts[AccountType.Main][currency],
              msgId,
              btn.arguments.slice(2, btn.arguments.length),
              tx,
              serverInfoState.prices[currency])
            )
          );
        } catch (e) {
          console.log('broadcast err: ' + e);
          return;
        }
        break;
      default:
        let msg = {
          id: 'answer-' + randomAsHex(32),
          value: btn.value,
          action: btn.action,
          args: btn.arguments,
          rows: [],
          timestamp: Date.now(),
          sender: globalState.profile!.id,
          receiver: chatInfo.id,
          hideBtn: true,
        };
        const timestamp = await backend.sendMsg({
          version: 1,
          value: btn.value,
          action: btn.action,
          receiver: chatInfo.id,
          args: btn.arguments,
        });
        if (timestamp == null) {
          return;
        }

        msg.timestamp = timestamp;

        dispatch(ChatsStore.actions.hideBtns({
          chatId: chatInfo.id,
          msgId: msgId,
        }));
        dispatch(ChatsStore.actions.addMessages([{
          chatId: chatInfo.id,
          msg: msg,
        }]));
        break;
    }
  };

  useEffect(() => {
    navigation.setOptions({
      title: stringUtils.formatNameOrAddress(user.title),
      headerRight: () => {
        return (
          <TouchableOpacity
            onPress={() =>
              navigation.navigate('ProfileInfo', {
                userId: chatInfo.id,
              })
            }
          >
            {user.isAddressOnly ?
              <WalletLogo
                currency={(user.value as AddressOnly).currency}
                size={45}
              /> :
              <Image
                source={{
                  uri: backend.getImgUrl((user.value as Profile).id, (user.value as Profile).lastUpdate),
                }}
                width={45}
                height={45}
                style={{ width: 45, height: 45, borderRadius: 25 }}
              />
            }
          </TouchableOpacity>
        );
      },
      headerRightContainerStyle: {
        marginRight: 20,
      },
    });
  }, []);

  useEffect(() => {
    const newMessages =
      Object.keys(chatsState.chats[chatInfo.id].messages)
      .map((key) => chatsState.chats[chatInfo.id].messages[key])
      .sort((a, b) => b.timestamp - a.timestamp);

    setMessages(newMessages);
    setLengthOffset(messages.length === 0 ? 0 : lengthOffset + (messages.length - newMessages.length));
  }, [chatsState.chats[chatInfo.id].messages]);

  useEffect(() => {
    if (chatInfo.notificationCount !== 0) {
      dispatch(ChatsStore.actions.removeNotification(chatInfo.id));
    }
  }, [chatInfo.notificationCount]);

  const renderItem = ({item, index}: {item: Message; index: number}) => {
    let line;
    if (notificationCount !== 0 && index === notificationCount - lengthOffset - 1) {
      line = (
        <View
          style={{
            marginTop: 10,
            marginBottom: 10,
            flexDirection: 'row',
            alignItems: 'center',
            justifyContent: 'center',
          }}>
          <Text style={styles.newMsgText}>
            {StringUtils.texts.UnreadMessagesTitle}
          </Text>
        </View>
      );
    }

    return (
      <View
        key={item.id}
        style={{
        paddingLeft: 15,
        paddingRight: 15,
        scaleY: -1,
      }}>
        {line}
        {
          item.action === '/tx' &&
          item.args.length >= 2 &&
          chatsState.transactions[toCurrency(item.args[0])].transactionById[item.args[1]] ?
            (() => {
              const tx = chatsState.transactions[toCurrency(item.args[0])].transactionById[item.args[1]];
              return <PaymentMsg
                onPress={() =>
                  navigation.navigate('TransactionDetails', {
                    transaction: tx,
                  })}
                tx={{
                  id: tx.id,
                  hash: tx.hash,
                  fullValue: tx.fullValue,
                  userId: tx.userId,
                  address: tx.address,
                  action: tx.action,
                  currency: tx.currency,
                  txType: tx.txType,
                  timestamp: tx.timestamp,
                  value: tx.value,
                  planckValue: tx.planckValue,
                  usdValue: tx.usdValue,
                  fee: tx.fee,
                  planckFee: tx.planckFee,
                  usdFee: tx.usdFee,
                  status: tx.status,
                }} />;
            })() :
            (<MessageView key={item.id} message={item} isOwner={item.sender === globalState.profile!.id}
                          onPressChatBtn={onPressChatBtn} />)
        }
      </View>
    );
  };

  return (
    <View style={styles.chat}>
      {<FlatList
        style={[{scaleY: -1 }, styles.messages]}
        getItemLayout={ (data, index) => (
          { length: 50, offset: 50 * index, index }
        )}
        windowSize={5}
        data={messages}
        renderItem={renderItem}
        keyExtractor={(item) => item.id}
        ListHeaderComponent={<View style={{marginBottom: user.isAddressOnly || !(user.value as Profile).isChatBot ? 80 : 50}}  />}
      />}
      {((user.isAddressOnly || !(user.value as Profile).isChatBot) &&
        <TouchableOpacity
          style={styles.sendBox}
          onPress={() =>
            !user.isAddressOnly
              ? navigation.navigate('SelectWallet', {
                chatId: chatInfo.id,
              })
              : navigation.navigate('Send', {
                isEditable: false,
                chatId: chatInfo.id,
                currency: (user.value as AddressOnly).currency,
              })
          }>
          <Image
            source={require('assets/img/send.png')}
            style={{width: 16, height: 25}}
          />
        </TouchableOpacity>
      )}
      {(!user.isAddressOnly && (user.value as Profile).isChatBot) && messages.length == 0 &&
        <TouchableOpacity
          style={styles.sendBox}
          onPress={() => {
            const msg = {
              id: 'answer-' + randomAsHex(32),
              value: 'Start',
              action: '/init',
              args: [],
              rows: [],
              timestamp: Date.now(),
              sender: globalState.profile!.id,
              receiver: chatInfo.id,
              hideBtn: true,
            };
            backend.sendMsg({
              version: 1,
              value: msg.value,
              action: msg.action,
              receiver: chatInfo.id,
              args: msg.args,
            }).then((timestamp) => {
              if (timestamp != null) {
                msg.timestamp = timestamp;
                dispatch(ChatsStore.actions.addMessages([{
                  chatId: chatInfo.id,
                  msg: msg,
                }]));
              }
            });
          }}
        >
          <Text style={styles.startBtnText}>Start</Text>
        </TouchableOpacity>
      }
    </View>
  );
};

const styles = StyleSheet.create({
  buttonText: {
    fontSize: 17,
    fontFamily: 'Roboto-Regular',
    fontStyle: 'normal',
    fontWeight: 'normal',
    color: 'white',
  },
  sendBox: {
    position: 'absolute',
    right: 10,
    bottom: 10,
    height: 60,
    width: 60,
    borderRadius: 50,
    backgroundColor: '#2AB2E2',
    alignItems: 'center',
    justifyContent: 'center',
  },
  chat: {
    flex: 1,
    alignItems: 'center',
    flexDirection: 'column',
  },
  messages: {
    width: '100%',
  },
  dividingLine: {
    width: '100%',
    height: 0.5,
    backgroundColor: '#C5C5C5',
  },
  newMsgDividingLine: {
    width: '15%',
    height: 1,
    borderColor: '#2AB2E2',
    borderBottomWidth: 1,
  },
  newMsgText: {
    textAlign: 'center',
    width: '100%',
    backgroundColor: '#51caf5',
    fontSize: 15,
    fontFamily: 'Roboto-Regular',
    fontStyle: 'normal',
    fontWeight: 'normal',
    color: 'white',
  },
  startBtnText: {
    fontSize: 15,
    fontFamily: 'Roboto-Regular',
    fontStyle: 'normal',
    fontWeight: 'normal',
    color: 'white',
  },
});
