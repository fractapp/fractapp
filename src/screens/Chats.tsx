import React, {useEffect} from 'react';
import {
  FlatList,
  StyleSheet,
  TouchableHighlight,
  TouchableOpacity,
  View,
} from 'react-native';
import {ChatShortInfo} from 'components/ChatShortInfo';
import {ChatInfo} from 'types/chatInfo';
import Ionicons from 'react-native-vector-icons/Ionicons';
import { useDispatch, useSelector } from 'react-redux';
import ChatsStore from 'storage/Chats';
import UsersStore from 'storage/Users';
// @ts-ignore
import {MAIN_BOT_ID} from '@env';

/**
 * Chat list screen
 * @category Screens
 */
export const Chats = ({navigation}: {navigation: any}) => {
  const dispatch = useDispatch();
  const chatsState: ChatsStore.State = useSelector((state: any) => state.chats);
  const usersState: UsersStore.State = useSelector((state: any) => state.users);

  useEffect(() => {
    navigation.setOptions({
      headerRight: () => {
        return (
          <TouchableOpacity onPress={() => navigation.navigate('Search')}>
            <Ionicons name="add" size={32} color="#2AB2E2" />
          </TouchableOpacity>
        );
      },
    });
  }, []);

  const getChats = () => {
    const chats = Object.keys(chatsState.chatsInfo).map((key) => chatsState.chatsInfo[key])
      .filter((value) => value.lastMsgId != null)
      .sort((a, b) => {
          const msgOne = chatsState.chats[a.id]!.messages[a.lastMsgId!]!;
          const msgTwo = chatsState.chats[b.id]!.messages[b.lastMsgId!]!;
          if (msgOne.sender === MAIN_BOT_ID) {
            return -1;
          } else if (msgTwo.sender === MAIN_BOT_ID) {
            return 1;
          }

          return msgTwo.timestamp - msgOne.timestamp;
        }
      );

    return chats;
  };

  const renderItem = ({item}: {item: ChatInfo}) => {
    let user = usersState.users[item.id]!;

    if (!usersState.users[item.id]) {
      user = {
        isAddressOnly: false,
        title: 'Deleted',
        value: {
          id: item.id,
          name: 'Deleted',
          username: 'Deleted',
          avatarExt: '',
          lastUpdate: 0,
          addresses: null,
          isChatBot: false,
        },
      };
      dispatch(UsersStore.actions.setUsers([user]));
    }
    const msg = chatsState.chats[item.id]!.messages[item.lastMsgId!]!;

    return (
      <TouchableHighlight
        onPress={() => navigation.navigate('Chat', {chatId: item.id})}
        underlayColor="#f8f9fb">
        <ChatShortInfo
          notificationCount={item.notificationCount}
          message={msg.value}
          timestamp={msg.timestamp}
          user={user ?? {
            isAddressOnly: false,
            value: {
              id: '0',
              name: 'Deleted',
              username: '-',
              avatarExt: '',
              lastUpdate: 0,
              addresses: new Map(),
            },
          }}
        />
      </TouchableHighlight>
    );
  };

  return (
    <View style={styles.chats}>
      <FlatList
        ItemSeparatorComponent={() => <View style={styles.dividingLine} />}
        showsVerticalScrollIndicator={false}
        style={styles.list}
        data={getChats()}
        renderItem={renderItem}
        keyExtractor={(item, index) => String(index)}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  chats: {
    alignItems: 'center',
    justifyContent: 'center',
    flex: 1,
  },
  list: {
    flex: 1,
    width: '100%',
  },
  dividingLine: {
    marginLeft: 80,
    width: '100%',
    height: 1,
    backgroundColor: '#f5f5f5',
  },
});
