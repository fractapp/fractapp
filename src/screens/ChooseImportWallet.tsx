import React, {useContext, useEffect} from 'react';
import {StyleSheet, View, Text, FlatList} from 'react-native';
import GlobalStore from 'storage/Global';
import {WhiteButton} from 'components/WhiteButton';
import backup from 'utils/backup';
import googleUtil from 'utils/google';

/**
 * Choose import wallet file import screen
 * @category Screens
 */
export const ChooseImportWallet = ({
  route,
  navigation,
}: {
  route: any;
  navigation: any;
}) => {
  const globalContext = useContext(GlobalStore.Context);

  const type: backup.BackupType = route.params.type;
  const wallets: Array<string> = route.params.wallets;
  const ids: Array<string> = route.params?.ids;

  const open = (v: string) => {
    globalContext.dispatch(GlobalStore.setLoading(true));
    if (type === backup.BackupType.File) {
      backup
        .getFile(v)
        .then((file) => {
          navigation.navigate('WalletFileImport', {
            file: file,
          });
          setTimeout(
            () => globalContext.dispatch(GlobalStore.setLoading(false)),
            500,
          );
        })
        .catch((e) => {
          globalContext.dispatch(GlobalStore.setLoading(false));
        });
    } else if (type === backup.BackupType.GoogleDrive) {
      googleUtil
        .getFileBackup(v)
        .then((file) => {
          navigation.navigate('WalletFileImport', {
            file: file,
          });
          setTimeout(
            () => globalContext.dispatch(GlobalStore.setLoading(false)),
            500,
          );
        })
        .catch((e) => {
          globalContext.dispatch(GlobalStore.setLoading(false));
        });
    }
  };

  useEffect(() => {
    globalContext.dispatch(GlobalStore.setLoading(false));
  }, []);

  return (
    <View
      style={{
        flexDirection: 'column',
        flex: 1,
        alignItems: 'center',
      }}>
      <Text style={styles.title}>Choose wallet</Text>

      <FlatList
        style={{
          flex: 1,
          marginTop: 30,
          width: '90%',
        }}
        data={wallets}
        renderItem={({item}) => {
          return (
            <View style={{marginBottom: 20}}>
              <WhiteButton
                text={item.replace('.json', '')}
                height={50}
                onPress={() => {
                  if (type === backup.BackupType.File) {
                    open(wallets[0]);
                  } else if (type === backup.BackupType.GoogleDrive) {
                    open(ids[0]);
                  }
                }}
              />
            </View>
          );
        }}
        keyExtractor={(item, index) => item + index}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  noResultsText: {
    marginTop: 10,
    fontSize: 19,
    fontFamily: 'Roboto-Regular',
    fontStyle: 'normal',
    fontWeight: 'normal',
    color: '#888888',
  },
  title: {
    marginTop: 80,
    fontSize: 25,
    fontFamily: 'Roboto-Regular',
    color: '#2AB2E2',
  },
  description: {
    textAlign: 'center',
    width: '90%',
    marginTop: 40,
    fontSize: 15,
    fontFamily: 'Roboto-Regular',
    color: '#888888',
  },
  newPassword: {
    marginTop: 30,
    width: '90%',
  },
  confirmPassword: {
    marginTop: 20,
    width: '90%',
  },
  error: {
    marginTop: 20,
    color: 'red',
    fontFamily: 'Roboto-Regular',
    fontSize: 15,
  },
});
