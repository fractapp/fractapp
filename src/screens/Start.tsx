import React from 'react';
import {StyleSheet, Text, View, Image} from 'react-native';
import {BlueButton} from 'components';

export const Start = ({navigation}: {navigation: any}) => {
  return (
    <View style={{flexDirection: 'column', flex: 1, alignItems: 'center'}}>
      <Image source={require('assets/img/logo.png')} style={styles.logo} />
      <Text style={styles.welcomeText}>Welcome to Fractapp</Text>
      <Text style={styles.description}>
        Messenger with cryptocurrency wallet
      </Text>
      <View style={styles.nextBtn}>
        <BlueButton
          text={'Start'}
          height={50}
          onPress={() => navigation.navigate('SettingWallet')}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  logo: {
    marginTop: 230,
    width: 61,
    height: 61,
    marginBottom: 10,
  },
  nextBtn: {
    width: '80%',
    position: 'absolute',
    bottom: 40,
  },
  welcomeText: {
    fontSize: 25,
    fontFamily: 'Roboto-Regular',
    color: '#2AB2E2',
  },
  description: {
    marginTop: 5,
    fontSize: 13,
    fontFamily: 'Roboto-Regular',
    color: '#888888',
  },
});
