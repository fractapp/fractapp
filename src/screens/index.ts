export {ImportWallet} from './ImportWallet';
export {SaveSeed} from './SaveSeed';
export {SaveWallet} from './SaveWallet';
export {SettingWallet} from './SettingWallet';
export {Start} from './Start';
export {WalletDetails} from './WalletDetails';
export {WalletFileBackup} from './WalletFileBackup';
export {Wallets} from './Wallets';
export {Settings} from './Settings';
export {ConfirmSaveSeed} from './ConfirmSaveSeed';
export {ImportSeed} from './ImportSeed';
export {WalletFileImport} from './WalletFileImport';
export {GoogleDrivePicker} from './GoogleDrivePicker';
export {Receive} from './Receive';
export {Milestone} from './Milestone';
export {TransactionDetails} from './TransactionDetails';
export {SplashScreen} from './SplashScreen';
export {Backup} from './Backup';
export {NewPassCode} from './NewPassCode';
export {VerifyPassCode} from './VerifyPassCode';
export {EditProfile} from './EditProfile';
export {EditPhoneNumber} from './EditPhoneNumber';
export {SelectCountry} from './SelectCountry';
export {ConfirmCode} from './ConfirmCode';
export {EditEmail} from './EditEmail';
export {Navigation} from './Navigation';
export {Chats} from './Chats';
export {Chat} from './Chat';
export {Connecting} from './Connecting';
export {EditUsername} from './EditUsername';
export {EditName} from './EditName';
export {Search} from './Search';
export {SelectWallet} from './SelectWallet';
export {Send} from './Send';
export {EnterAmount} from './EnterAmount';
export {SendWithEnterAddress} from './SendWithEnterAddress';
