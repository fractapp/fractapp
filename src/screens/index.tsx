export { default as ImportWallet } from './ImportWallet';
export { default as SaveSeed } from './SaveSeed';

export { default as SaveWallet } from './SaveWallet';
export { default as SettingWallet } from './SettingWallet';
export { default as Start } from './Start';

export { default as WalletDetails } from './WalletDetails';

export { default as WalletEncryption } from './WalletEncryption';
export { default as Wallets } from './Wallets';
export { ConfirmSaveSeed } from './ConfirmSaveSeed'