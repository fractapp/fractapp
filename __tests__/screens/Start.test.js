import React from 'react';
import {Start} from 'screens/Start';
import renderer from 'react-test-renderer';
import {render, fireEvent} from '@testing-library/react-native';
import StringUtils from 'utils/string';

jest.mock('react-native-i18n', () => ({
  t: (value) => value,
}));

it('Test positive', () => {
  const tree = renderer.create(<Start navigation={null} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test click start', () => {
  const mockFn = jest.fn();
  const component = render(<Start navigation={{navigate: mockFn}} />);
  fireEvent.press(component.getByText(StringUtils.texts.StartBtnTitle));
  expect(mockFn).toBeCalledWith('Legal');
});
