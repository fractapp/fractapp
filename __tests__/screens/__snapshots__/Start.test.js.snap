// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test positive 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../assets/img/logo.png",
      }
    }
    style={
      Object {
        "height": 61,
        "marginBottom": 10,
        "marginTop": 230,
        "width": 61,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#2AB2E2",
        "fontFamily": "Roboto-Regular",
        "fontSize": 25,
      }
    }
  >
    [missing "en.welcome_title" translation]
  </Text>
  <Text
    style={
      Object {
        "color": "#888888",
        "fontFamily": "Roboto-Regular",
        "fontSize": 13,
        "marginTop": 5,
      }
    }
  >
    [missing "en.welcome_description" translation]
  </Text>
  <View
    style={
      Object {
        "bottom": 40,
        "position": "absolute",
        "width": "80%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2AB2E2",
          "borderRadius": 10,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "Roboto-Regular",
            "fontSize": 15,
            "fontStyle": "normal",
            "fontWeight": "normal",
          }
        }
      >
        [missing "en.start_btn_title" translation]
      </Text>
    </View>
  </View>
</View>
`;
